internal/circonus/metric_filters.go

```diff
@@ -85,10 +67,100 @@ func (c *Check) defaultFilters() [][]string {
+    ["allow", "^pod_init_container_status_(terminated|waiting)_reason(_count)?$", "init containers health"],
+    ["allow", "^kube_deployment_(created|spec_replicas)$", "deployments"],
+    ["allow", "^kube_deployment_status_(replicas|replicas_updated|replicas_available|replicas_unavailable)$", "deployments"],
+    ["allow", "^kube_job_status_failed$", "health"],
+    ["allow", "^kube_persistentvolume_status_phase$", "health"],
+    ["allow", "^kube_hpa_(spec_max|status_current)_replicas$", "scale"],
+    ["allow", "^kube_pod_start_time$", "pods"],
+    ["allow", "^(kube_)?pod_status_phase(_count)?$", "tags", "and(or(phase:Running,phase:Pending,phase:Failed,phase:Succeeded))", "pods"],
+    ["allow", "^pod_status_phase$", "pods"],
+    ["allow", "^kube_pod_info$", "pods"],
+    ["allow", "^kube_(service|deployment)_labels$", "ksm inventory"],
+    ["allow", "^kube_node_spec_unschedulable$", "node status"],
+    ["allow", "^kube_node_status_allocatable$", "node status"],
+    ["allow", "^kube_node_status_condition$", "node status health"],
+    ["allow", "^kube_namespace_status_phase$", "tags", "namespaces"],
+    ["allow", "^utilization$", "utilization health"],
+    ["allow", "^kube_deployment_(metadata|status_observed)_generation$", "health"],
+    ["allow", "^kube_daemonset_status_(current|desired)_number_scheduled$", "health"],
+    ["allow", "^kube_statefulset_status_(replicas|replicas_ready)$", "health"],
+    ["allow", "^deployment_generation_delta$", "health"],
+    ["allow", "^daemonset_scheduled_delta$", "health"],
+    ["allow", "^statefulset_replica_delta$", "health"],
+    ["allow", "^coredns_(dns|forward)_request_(count_total|duration_seconds_avg)$", "dns health"],
+    ["allow", "^coredns_(dns|forward)_response_rcode_count_total$", "dns health"],
+    ["allow", "^kubedns*","dns health"],
+    ["allow", "^events$", "events"],
+    ["allow", "^collect_.*$", "agent collection stats"],
+    ["allow", "^authentication_attempts$", "api auth health"],
+    ["allow", "^cadvisor.*$", "cadvisor"],
+    ["deny", "^.+$", "all other metrics"]
+    ]
+}
+`
+)
+
+type metricFilters struct {
+	Filters [][]string `json:"metric_filters"`
+}
+
+func (c *Check) loadMetricFilters() [][]string {
+	mfConfigFile := viper.GetString(keys.MetricFiltersFile)
+	data, err := ioutil.ReadFile(mfConfigFile)
+	if err != nil {
+		c.log.Warn().Err(err).Str("metric_filter_config", mfConfigFile).Msg("using defaults")
+		return c.defaultFilters()
+	}
+
+	var mf metricFilters
+	if err := json.Unmarshal(data, &mf); err != nil {
+		c.log.Warn().Err(err).Str("metric_filter_config", mfConfigFile).Msg("using defaults")
+		return c.defaultFilters()
+	}
+
+	return mf.Filters
+}
+
+func (c *Check) defaultFilters() [][]string {
+
+	var defaultMetricFiltersData []byte
+
+	currversion, err := version.NewVersion(c.clusterVers)
+	if err != nil {
+		c.log.Warn().Err(err).Msg("parsing api version")
+		currversion, _ = version.NewVersion(v120str)
```

Comment:
Parse v120, as you do below, first and just assign currversion = v120 here. Then you won't have an unchecked error issue if you run a linter on this, and you remove a call to NewVersion(). I'm guessing that the intended behavior is to default to v1.20 filters.